onnxscript/ir/_core.py

```diff
@@ -1528,6 +1530,9 @@ def const_value(
                 raise TypeError(
                     f"Expected value to be a TensorProtocol or None, got '{type(value)}'"
                 )
+        if value is not None:
+            self.shape = value.shape
+            self.dtype = value.dtype
```

Comment:
Probably not possible because when we roundtrip from ONNX files, some models will have initializers but not ValueInfoProto for certain values. We need to respect the model file and not add valueinfo for these values. Probably good to leave it to some (micro) shape inference passes @titaiwangms